### PR TITLE
Render profile tab content when profile page is loaded via url. Fixes #11525

### LINKED ISSUE #11525: Profile pages have no information when opened as non-modal pages
If you open a player's profile screen in a window by itself, then you see the header and tabs but with no information shown:
screenshot

The correct behaviour is that the information from the "Profile" tab should appear.

You can reproduce this bug in any of these ways:
- Right-click on a player's name / avatar image and select the option to open in a new tab or window.
- Normal click on a player's name / avatar image and then reload the page.
- Click on a profile link that you find elsewhere such as in the wiki. Here's an example: https://habitica.com/profile/d904bd62-da08-416b-a816-ba797c9ee265

The same problem does not occur when you open a profile in a modal.

### PR DESCRIPTION
**Pull request by vamsijv**
[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #11525
 
**Description**:
This PR is related to issue#11525 where the profile page when accessed via its profile url renders blank content by default in the section beneath its three tabs - Profile, Stats, Achievements. Unlike here, this issue doesnt occur when profile page is loaded within the modal window that appears when an avatar of a user is clicked. As part of proposed solution, the fix aims to load the content from 'Profile' tab as default whenever the page content is loaded from its url.

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

Following components have been modified to address the issue:
**_profile.vue_**
This is child component that is used to fetch and render the user profile content when accessed in parent component.
The content was rendered blank because the 'selectedPage' was determined as 'undefined' upon the component load. This is because the 'startingPage' param/prop was not getting populated with desired value - 'profile'.
I have added 'profile' as fallback value for 'selectedPage' in case its not populated from either from this or its parent component.

**_profilePage.vue_**
This component loads when profile page is accessed via url - [Sample url](https://habitica.com/profile/d904bd62-da08-416b-a816-ba797c9ee265). This is parent component of the above *profile.vue* and it seems that existing 'startingPage' prop doesnt seem to be populated on the load, thus causing the issue. This component is now modified to have 'startingPage' as 'data' and is set with default value 'profile'.

The changes from either of the components is good enough to populate the content and resolve the issue, but I have handled at both ends to ensure no failure.

**Current Status:**
The fix has been developed and tested locally. However I could not run all the tests(npm run test) successfully in my machine due to timeout error. As such, **I'm hoping to receive the full test results of test suite as part of this PR creation before submitting it for actual review**. **I will update once its ready for review** or if I need more info on further activities.

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: 984e9826-a483-4036-beed-d94d2fbc9446

**_Before fix_**: Profile page on load
before fix

**_After fix_**: Profile page on load
after fix

### CODE REVIEW

**Comment by vamsijv** [comment]
The result of test suite execution during PR creation came positive. Dev and testing are complete. The **PR is ready for review**. Please share your thoughts and let me know in case of queries or next steps.

File: website/client/src/components/userMenu/profilePage.vue [hunk 1]
```diff
@@ -24,6 +24,11 @@ export default {
   components: {
     profile,
   },
-  props: ['userId', 'startingPage'],
```
Comment (paglias) [inline website/client/src/components/userMenu/profilePage.vue:27]: the startingPage prop is still used in some places so it cannot be removed, what you can do is having a default value for it https://vuejs.org/v2/guide/components-props.html

**Review by paglias — CHANGES_REQUESTED** [review]

File: website/client/src/components/userMenu/profilePage.vue (hunk 1 above)
Comment (vamsijv) [reply website/client/src/components/userMenu/profilePage.vue:27]: Before removing 'startingPage' as prop from profilePage.vue (as part of my original changes), I had searched in source code for any references to '<profilePage' which returned 0 results leading to my understanding that profilePage.vue is not used as child component and has no existing parent component. So I removed it as I thought props would be set generally from parent component. Guess I m wrong as you mentioned its being used somewhere else. Pardon my limited knowledge on this as I m a newbie to vue.js, and if I'm missing something obvious.

I have made your requested changes and validated locally and pushed the commit to my fork repo. Please review. Thanks.

Comment (paglias) [reply website/client/src/components/userMenu/profilePage.vue:27]: No worries! The profilePage component is used as <profile-page> in HTML code, that’s why you couldn’t find it.

I’ll give a look at this tomorrow and merge it if everything looks good on Friday.

Thanks!

**Comment by vamsijv** [comment]
This PR is open for review

**Comment by vamsijv** [comment]
Can we please have an update here

**Comment by paglias** [comment]
Thanks @vamsijv and sorry for the delay, the fix seems to be working well. It'll be merged the next time we merge contributors PRs (today or tomorrow)

**Review by paglias — APPROVED** [review]

**Comment by paglias** [comment]
Thanks @vamsijv ! I've notated this PR towards your 2nd tier

**Comment by vamsijv** [comment]
Cool. Appreciate it!!

**Pull request merged**